lib/src/json_schema/utils.dart

```diff
@@ -106,3 +106,20 @@ class DefaultValidators {
     }
   }
 }
+
+extension DeepMerge<K> on Map<K, dynamic> {
+  /// Merge other into this with properties in this taking precedence.
+  /// Lists ARE NOT concatenated together.
+  Map<K, dynamic> deepMerge(Map<K, dynamic> other) {
+    other.forEach((key, value) {
+      if (this.containsKey(key)) {
+        if (value is Map && this[key].runtimeType == value.runtimeType) {
```

Comment:
Maybe I'm missing the point of this check but are you basically trying to determine of the original value was a map? If so I'd do instead
```suggestion
        if (value is Map && this[key] is Map) {
```

Reply:
I'm also trying to check to make sure the 2 maps are of the same type. But perhaps they are always `Map<String, dynamic>`?

Reply:
Ah, I see. That's probably worthwhile. I'm a little concerned that the runtimeType string might not always be equal when you have different map implementations though...

Reply:
`Map.unmodifiable` specifically, comes to mind. https://api.flutter.dev/flutter/dart-core/Map/Map.unmodifiable.html Although I'm not sure if the library currently would handle unmodifiable maps well anyway.

Reply:
oofff.... In that case we would need to make a new list instead of updating the list. Is that something we should take into consideration?